### PR TITLE
various changes

### PR DESCRIPTION
**Pull request by dsprenkels**
- made some textual improvements to documentation
- shortened some rust code
- remove unneeded (since 99022ed) function decorator

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -383,7 +383,7 @@ In terms of writing code:
 3. `chan` does not expose `len` or `cap` methods. (For no reason other than
    to start with a totally minimal API. In particular, calling `len` or `cap`
    on a channel is often The Wrong Thing. But not always. So this restriction
-   will probably be lifted.)
+   may lifted in the future.)
```
Comment (BurntSushi) [inline src/lib.rs]: s/may lifted/may be lifted

**Comment by BurntSushi** [comment]
One little nit, but otherwise looks good, thank you!

**Comment by dsprenkels** [comment]
Resolved the comment. Should be alright now.

**Comment by BurntSushi** [comment]
Thanks!

**Pull request merged**